ui/app/components/clients/sync-bar-chart.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { formatNumbers, formatTooltipNumber } from 'vault/utils/chart-helpers';
+import type { SerializedChartData } from 'vault/client-counts';
+import { tracked } from '@glimmer/tracking';
+import { parseAPITimestamp } from 'core/utils/date-formatters';
+import { format } from 'date-fns';
+
+interface Args {
+  data: SerializedChartData[];
+  dataKey: string;
+  chartTitle: string;
+  chartHeight?: number;
+}
+
+interface ChartData {
+  x: string;
+  y: number | null;
+  tooltip: string;
+  legendX: string;
+  legendY: string;
+}
+
+/**
+ * @module ClientsSyncBarChartComponent
+ * Renders a bar chart of secret syncs over time.
+ *
+ * @example
+ * ```js
+ * <Clients::SyncBarChart
+    @chartTitle="Secret Sync client counts"
+    @data={{this.model}}
+    @dataKey="secret_syncs"
+    @showTable={{true}}
+    @chartHeight={{200}}
+  />
+ * ```
+ */
+export default class ClientsSyncBarChartComponent extends Component<Args> {
+  barWidth = 7;
```

Comment:
Since this same const is defined in `vertical-bar-chart` would it be useful to move to `vault/utils/chart-helpers` and import from there?